### ISSUE TITLE
accurately finding correct .so version when multiple versions are installed

### ISSUE THREAD

**Issue by soumith** [soumith/cudnn.torch#75]

**Comment by soumith** [comment]
@Alexey-Kamenev if you have multiple versions installed, I'd say check libcudnn.so and which version it is linked to.

**Comment by Alexey-Kamenev** [comment]
Currently I have this:

alexeyk@gcrgpu1071:$ ll /usr/local/cuda-7.0/lib64/libcudn*
lrwxrwxrwx 1 alexeyk alexeyk  15 Aug 21 09:45 /usr/local/cuda-7.0/lib64/libcudnn.so -> libcudnn.so.7.0
lrwxrwxrwx 1 alexeyk alexeyk  18 Aug 21 09:45 /usr/local/cuda-7.0/lib64/libcudnn.so.7.0 -> libcudnn.so.7.0.64
-rwxr-xr-x 1 alexeyk alexeyk 46M Aug 21 09:45 /usr/local/cuda-7.0/lib64/libcudnn.so.7.0.64
-rw-r--r-- 1 alexeyk alexeyk 46M Aug 21 09:45 /usr/local/cuda-7.0/lib64/libcudnn_static.a

Not sure if my LD_LIBRARY_PATH is somehow messed up...

**Comment by borisfom** [comment]
Your installation is just old - please update the package from Nvidia repo:

bfomitchev@borisf-desktop:~/git/caffe$ sudo dpkg -L libcudnn4
/.
/usr
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/libcudnn.so.4.0.3
/usr/share
/usr/share/doc
/usr/share/doc/libcudnn4
/usr/share/doc/libcudnn4/copyright
/usr/share/doc/libcudnn4/changelog.gz
/usr/share/lintian
/usr/share/lintian/overrides
/usr/share/lintian/overrides/libcudnn4
/usr/lib/x86_64-linux-gnu/libcudnn.so.4

**Comment by borisfom** [comment]
To find which libcudnn your particular Torch installation uses, look at cudnn/ffi.lua, it should have a line like : 
local libnames = {'libcudnn.so.4', 'libcudnn.4.dylib'}

**Comment by soumith** [comment]
@borisfom no, R4 has not become the default yet, as it is still an RC* release. Alexey has cudnn R3 installed correctly, and the bindings for it as well.

**Comment by soumith** [comment]
@Alexey-Kamenev I dont really know what to suggest here. The same version of cudnn, and same set of bindings, seem to work for me

**Comment by soumith** [comment]
Alexey, could you double-check the LD_LIBRARY_PATH.
Also, can you add a:

print(cudnn.version) 

and see if the current version is being loaded.
